diff -up libtermkey-0.22/Makefile.omv~ libtermkey-0.22/Makefile
--- libtermkey-0.22/Makefile.omv~	2023-09-06 22:51:18.572137216 +0100
+++ libtermkey-0.22/Makefile	2023-09-06 22:57:39.463192595 +0100
@@ -1,15 +1,5 @@
 pkgconfig = $(shell PKG_CONFIG_PATH=$(PKG_CONFIG_PATH) pkg-config $(1))
 
-ifeq ($(shell uname),Darwin)
-  LIBTOOL ?= glibtool
-else
-  LIBTOOL ?= libtool
-endif
-
-ifneq ($(VERBOSE),1)
-  LIBTOOL +=--quiet
-endif
-
 override CFLAGS +=-Wall -std=c99
 
 ifeq ($(DEBUG),1)
@@ -34,8 +24,10 @@ else
   override LDFLAGS+=-lncurses
 endif
 
-OBJECTS=termkey.lo driver-csi.lo driver-ti.lo
-LIBRARY=libtermkey.la
+override CFLAGS += -fPIC
+
+OBJECTS=termkey.o driver-csi.o driver-ti.o
+LIBRARY=libtermkey.so.1.$(VERSION_AGE).$(VERSION_REVISION)
 
 DEMOS=demo demo-async
 
@@ -67,28 +59,28 @@ MAN7DIR=$(MANDIR)/man7
 all: $(LIBRARY) $(DEMOS)
 
-%.lo: %.c termkey.h termkey-internal.h
-	$(LIBTOOL) --mode=compile --tag=CC $(CC) $(CFLAGS) -o $@ -c $<
+%.o: %.c termkey.h termkey-internal.h
+	$(CC) $(CFLAGS) -o $@ -c $<
 
 $(LIBRARY): $(OBJECTS)
-	$(LIBTOOL) --mode=link --tag=CC $(CC) -rpath $(LIBDIR) -version-info $(VERSION_CURRENT):$(VERSION_REVISION):$(VERSION_AGE) $(LDFLAGS) -o $@ $^
+	$(CC) $(LDFLAGS) -shared -Wl,-soname,libtermkey.so.1 -o $@ $^
 
-demo: $(LIBRARY) demo.lo
-	$(LIBTOOL) --mode=link --tag=CC $(CC) -o $@ $^
+demo: $(LIBRARY) demo.o
+	$(CC) -o $@ $^
 
-demo-async: $(LIBRARY) demo-async.lo
-	$(LIBTOOL) --mode=link --tag=CC $(CC) -o $@ $^
+demo-async: $(LIBRARY) demo-async.o
+	$(CC) -o $@ $^
 
 demo-glib.lo: demo-glib.c termkey.h
-	$(LIBTOOL) --mode=compile --tag=CC $(CC) -o $@ -c $< $(call pkgconfig, glib-2.0 --cflags)
+	$(CC) -o $@ -c $< $(call pkgconfig, glib-2.0 --cflags)
 
-demo-glib: $(LIBRARY) demo-glib.lo
-	$(LIBTOOL) --mode=link --tag=CC $(CC) -o $@ $^ $(call pkgconfig, glib-2.0 --libs)
+demo-glib: $(LIBRARY) demo-glib.o
+	$(CC) -o $@ $^ $(call pkgconfig, glib-2.0 --libs)
 
-t/%.t: t/%.c $(LIBRARY) t/taplib.lo
-	$(LIBTOOL) --mode=link --tag=CC $(CC) -o $@ $^
+t/%.t: t/%.c $(LIBRARY) t/taplib.o
+	$(CC) -o $@ $^
 
 t/taplib.lo: t/taplib.c
-	$(LIBTOOL) --mode=compile --tag=CC $(CC) $(CFLAGS) -o $@ -c $^
+	$(CC) $(CFLAGS) -o $@ -c $^
 
 .PHONY: test
 test: $(TESTFILES)
@@ -96,17 +88,16 @@ test: $(TESTFILES)
 
 .PHONY: clean-test
 clean-test:
-	$(LIBTOOL) --mode=clean rm -f $(TESTFILES) t/taplib.lo
+	rm -f $(TESTFILES) t/taplib.o
 
 .PHONY: clean
 clean: clean-test
-	$(LIBTOOL) --mode=clean rm -f $(OBJECTS) $(DEMO_OBJECTS)
-	$(LIBTOOL) --mode=clean rm -f $(LIBRARY)
-	$(LIBTOOL) --mode=clean rm -rf $(DEMOS)
+	rm -f $(OBJECTS) $(DEMO_OBJECTS)
+	rm -f $(LIBRARY)
+	rm -rf $(DEMOS)
 
 .PHONY: install
 install: install-inc install-lib install-man
-	$(LIBTOOL) --mode=finish $(DESTDIR)$(LIBDIR)
 
 install-inc: termkey.h
 	install -d $(DESTDIR)$(INCDIR)
@@ -116,7 +107,9 @@ install-inc: termkey.h
 
 install-lib: $(LIBRARY)
 	install -d $(DESTDIR)$(LIBDIR)
-	$(LIBTOOL) --mode=install install libtermkey.la $(DESTDIR)$(LIBDIR)/libtermkey.la
+	install -c -m 755 $(LIBRARY) $(DESTDIR)$(LIBDIR)/
+	ln -sf $(LIBRARY) $(DESTDIR)$(LIBDIR)/libtermkey.so.1
+	ln -sf $(LIBRARY) $(DESTDIR)$(LIBDIR)/libtermkey.so
 
 install-man:
 	install -d $(DESTDIR)$(MAN3DIR)
